packages/platform-sdk-profiles/src/drivers/memory/profiles/services/coin.factory.ts

```diff
@@ -0,0 +1,47 @@
+import { Coins } from "@arkecosystem/platform-sdk";
+
+import { IProfile } from "../../../../contracts";
+import { ICoinFactory } from "../../../../contracts/services/coin-factory";
+import { container } from "../../../../environment/container";
+import { Identifiers } from "../../../../environment/container.models";
+
+export class CoinFactory implements ICoinFactory {
+	readonly #profile: IProfile;
+
+	public constructor(profile: IProfile) {
+		this.#profile = profile;
+	}
+
+	public make(coin: string, network: string, options: object = {}, useForce = false): Coins.Coin {
+		if (this.#profile.usesCustomPeer() && this.#profile.peers().has(coin, network)) {
+			return this.createDriver(
+				coin,
+				network,
+				{
+					peer: this.#profile.peers().getRelay(coin, network)?.host,
+					peerMultiSignature: this.#profile.peers().getMultiSignature(coin, network)?.host,
+				},
+				true,
+			);
+		}
+
+		return this.createDriver(coin, network);
+	}
+
+	private createDriver(coin: string, network: string, options: object = {}, useForce = false): Coins.Coin {
+		if (!useForce && this.#profile.coins().has(coin, network)) {
+			return this.#profile.coins().get(coin, network);
+		}
+
+		const instance = Coins.CoinFactory.make(container.get<Coins.CoinSpec>(Identifiers.Coins)[coin.toUpperCase()], {
+			network,
+			httpClient: container.get(Identifiers.HttpClient),
+			...options,
+		});
+
+		// @TODO: we should move this out but that means potentially more repetitive work for the client
```

Comment:
I'm torn on what to do here. This is just a factory and shouldn't store the coin but otherwise the client will have to do more work every time they need to create a coin instance.

I think the best solution here would be to create a separate function or use case that combines the factory and repository so that the client doesn't have to deal with the storage of the coin.